lib/lyber_core/robot.rb

```diff
@@ -1,109 +1,99 @@
 # frozen_string_literal: true
 
-require 'benchmark'
-require 'socket'
-require 'active_support'
-require 'active_support/core_ext/object/blank' # String#blank?
-
 module LyberCore
-  module Robot
-    # Add the ClassMethods to the class this is being mixed into
-    def self.included(base)
-      base.extend ClassMethods
-    end
+  # Base class for all robots
+  class Robot
+    include Sidekiq::Job
 
-    module ClassMethods
-      # Called by job-manager on derived-class
-      # Instantiate the Robot and call #work with the passed in druid
-      def perform(druid, *context)
-        bot = new
-        bot.work druid, context
-      end
-    end
+    attr_reader :workflow_name, :process, :druid
 
-    attr_accessor :check_queued_status
-    attr_reader :workflow_name, :process
+    delegate :lane_id, to: :workflow
 
-    def initialize(workflow_name, process, check_queued_status: true)
-      Signal.trap('QUIT') { puts "#{Process.pid} ignoring SIGQUIT" } # SIGQUIT ignored to let the robot finish
+    def initialize(workflow_name, process)
       @workflow_name = workflow_name
       @process = process
-      @check_queued_status = check_queued_status
     end
 
     def workflow_service
-      raise 'The workflow_service method must be implemented on the class that includes LyberCore::Robot'
+      @workflow_service ||= WorkflowClientFactory.build
```

Comment:
I'm going to trying providing the workflow client here instead of requiring it to be provided by subclasses.